src/sediment_transport/overland_flow_transport.jl

```diff
@@ -0,0 +1,270 @@
+abstract type AbstractSedimentLandTransportModel{T} end
+
+## Total sediment transport in overland flow structs and functions
+@get_units @grid_loc @with_kw struct SedimentLandTransportVariables{T}
+    # Total sediment flux
+    amount::Vector{T} | "t dt-1"
+    deposition::Vector{T} | "t dt-1"
+end
+
+function SedimentLandTransportVariables(
+    n;
+    amount::Vector{T} = fill(mv, n),
+    deposition::Vector{T} = fill(mv, n),
+) where {T}
+    return SedimentLandTransportVariables{T}(; amount = amount, deposition = deposition)
+end
+
+@get_units @grid_loc @with_kw struct SedimentLandTransportBC{T}
+    # Eroded material
+    erosion::Vector{T} | "t dt-1"
+    # Transport capacity
+    transport_capacity::Vector{T} | "t dt-1"
+end
+
+function SedimentLandTransportBC(
+    n;
+    erosion::Vector{T} = fill(mv, n),
+    transport_capacity::Vector{T} = fill(mv, n),
+) where {T}
+    return SedimentLandTransportBC{T}(;
+        erosion = erosion,
+        transport_capacity = transport_capacity,
+    )
+end
+
+@with_kw struct SedimentLandTransportModel{T} <: AbstractSedimentLandTransportModel{T}
+    boundary_conditions::SedimentLandTransportBC{T}
+    variables::SedimentLandTransportVariables{T}
+end
+
+function SedimentLandTransportModel(inds)
+    n = length(inds)
+    vars = SedimentLandTransportVariables(n)
+    bc = SedimentLandTransportBC(n)
+    model = SedimentLandTransportModel(; boundary_conditions = bc, variables = vars)
+    return model
+end
+
+function update_boundary_conditions!(
+    model::SedimentLandTransportModel,
+    erosion_model::SoilErosionModel,
+    transport_capacity_model::AbstractTransportCapacityModel,
+)
+    (; erosion, transport_capacity) = model.boundary_conditions
+    (; amount) = erosion_model.variables
+    @. erosion = amount
+
+    (; amount) = transport_capacity_model.variables
+    @. transport_capacity = amount
+end
+
+function update!(model::SedimentLandTransportModel, network)
+    (; erosion, transport_capacity) = model.boundary_conditions
+    (; amount, deposition) = model.variables
+
+    accucapacityflux!(amount, erosion, network, transport_capacity)
+    deposition .= erosion
+end
+
+## Total transport capacity with particle differentiation structs and functions
+@get_units @grid_loc @with_kw struct SedimentLandTransportDifferentiationVariables{T}
```

Comment:
Any suggestion? I'm afraid if I don't include sediment in the name then it may get confusing with the hydrology (``LandTransport`` or ``RiverTransport``) or you think that would be okay? Or maybe I could use part or particle instead of differentiation ``SedimentLandTransportParticle``? Does not shorten that much though and I like that the names are very explicit.

Reply:
Let's stick to the name then!